### DISCUSSION TITLE
Enabling FTP - ERR error loading required template: open templates/webadmin/base.html: no such file or directory panic

### DISCUSSION THREAD

**Discussion by davecowley** [drakkan/sftpgo#673] [Q&A] [answered]
Hi, im not sure if I am being dumb or not but I seem to be having issues enabling FTPD.  SFTP works fine and I want to run both services.  I have modifed sftpgo.json but only in 2 fields.  
I set the port to be 21 and then set enable_site to be true.

 "ftpd": {
    "bindings": [
      {
        "port": 21,
        "address": "",
        "apply_proxy_config": true,
        "tls_mode": 0,
        "force_passive_ip": "",
        "passive_ip_overrides": [],
        "client_auth_type": 0,
        "tls_cipher_suites": [],
        "passive_connections_security": 0,
        "active_connections_security": 0,
        "debug": false
      }
    ],
    "banner": "",
    "banner_file": "",
    "active_transfers_port_non_20": true,
    "passive_port_range": {
      "start": 50000,
      "end": 50100
    },
    "disable_active_mode": false,
    "enable_site": true,
    "hash_support": 0,
    "combine_support": 0,
    "certificate_file": "",

When I do these changes the webgui on 8080 won't load 

I also get this when trying to run the service.

2022-01-12T14:51:24.233 INF No host keys configured and "id_rsa" does not exist; try to create a new host key
2022-01-12T14:51:24.240 ERR error loading required template: open templates/webadmin/base.html: no such file or directory
panic: open templates/webadmin/base.html: no such file or directory

goroutine 123 [running]:
github.com/drakkan/sftpgo/v2/util.LoadTemplate(0xc00013f0c0, {0xc000557130, 0x16f0375, 0xf})
        github.com/drakkan/sftpgo/v2/util/util.go:308 +0x16e
github.com/drakkan/sftpgo/v2/httpd.loadAdminTemplates({0xc000309db6, 0x9})
        github.com/drakkan/sftpgo/v2/httpd/webadmin.go:311 +0x1d1c
github.com/drakkan/sftpgo/v2/httpd.(*Conf).Initialize(0xc000478640, {0x1a27688, 0x1})
        github.com/drakkan/sftpgo/v2/httpd/httpd.go:471 +0x2f7
github.com/drakkan/sftpgo/v2/service.(*Service).startServices.func2()
        github.com/drakkan/sftpgo/v2/service/service.go:185 +0x3f
created by github.com/drakkan/sftpgo/v2/service.(*Service).startServices
        github.com/drakkan/sftpgo/v2/service/service.go:184 +0x3c8
admin@asx-wh-xxx1:~$

Its on a fresh install of ubuntu, new install of SFTPgo via pp and the only other service I have running is webmin.

Could you point me in the right direction please? Have I enabled ftpd the right way?

Thanks
Dave

**Comment by drakkan** [comment, marked as answer]
Hi,

please set a port such 2121, to listen on port 21 you need to execute SFTPGo as root user or to enable the correct capability, take a look [here](https://github.com/drakkan/sftpgo/blob/main/docs/full-configuration.md#binding-to-privileged-ports) for details.

I guess you get the second error while trying to start SFTPGo manually (and so the correct config file is not used), please use the systemd unit:

```
$ systemctl status sftpgo.service
$ sudo systemctl start sftpgo.service
```

to see the relevant logs please use `journalctl -f`

**Comment by davecowley** [comment]
as simple as that.  thank you, I have been going round in a circle the past couple of days.  I had tried a different port but failed.

thanks again
Dave

**Reply by drakkan** [reply]
> as simple as that. thank you, I have been going round in a circle the past couple of days. I had tried a different port but failed.
> 
> thanks again Dave

No problem, thanks for using (or trying to use) SFTPGo
